src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_depth2img.py

```diff
@@ -25,36 +23,15 @@
 from ...configuration_utils import FrozenDict
 from ...models import AutoencoderKL, UNet2DConditionModel
 from ...schedulers import KarrasDiffusionSchedulers
-from ...utils import PIL_INTERPOLATION, deprecate, is_accelerate_available, logging, randn_tensor
-from ..pipeline_utils import DiffusionPipeline, ImagePipelineOutput
+from ...utils import deprecate, logging
+from ..pipeline_utils import ImagePipelineOutput
 
+from .pipeline_stable_diffusion_img2img import StableDiffusionImg2ImgPipeline, preprocess
 
 logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
 
 
-# Copied from diffusers.pipelines.stable_diffusion.pipeline_stable_diffusion_img2img.preprocess
```

Comment:
:sweat_smile: why do we delete everything here?

Reply:
I thought it would be good to import the preprocess function directly from the img2img pipeline file to make it easier to update/review the code in one place.
The `preprocess` function is identical for several files, like pipeline_stable_diffusion_instruct_pix2pix.py, pipeline_stable_diffusion_latent_upscale.py, pipeline_stable_diffusion_upscale.py, pipeline_stable_diffusion_depth2img.py. 
pipeline_stable_diffusion_inpaint_legacy.py also has a very similar  `preprocess_image` function that does the same with fewer type checks.

Maybe move the `preprocess` function to a separate utils file that the pipelines import from, as a way to standardize inputs received by pipelines. Lmk what you think!